classes/ObjectModel.php

```diff
@@ -1229,14 +1254,19 @@ public function validateField($field, $value, $id_lang = null, $skip = [], $huma
 
             if (!empty($value)) {
                 $res = true;
-                if (Tools::strtolower($data['validate']) == 'iscleanhtml') {
+                if (Tools::strtolower($data['validate']) === 'iscleanhtml') {
                     if (!call_user_func(['Validate', $data['validate']], $value, $ps_allow_html_iframe)) {
                         $res = false;
                     }
                 } else {
                     if (!call_user_func(['Validate', $data['validate']], $value)) {
                         $res = false;
                     }
+                    if (Tools::strtolower($data['validate']) === 'isint') {
+                        if ($value < StockSettings::INT_32_MAX_NEGATIVE || $value > StockSettings::INT_32_MAX_POSITIVE) {
+                            $res = false;
+                        }
+                    }
```

Comment:
```suggestion
```

Reply:
The `size` option is not adapted to integer fields so it shouldn't test this here Besides we shouldn't couple ObjectModel with stock specific values and `StockSettings`

The `isint` validate method doesn't tell us what kind of integer the DB is using, it can be TINYINT, BIGINT, UNSIGNED BIGINT, INT, ... Each of these columns types have different ranges, we could introduce an `integer_type` option that would automatically deduce the corresponding ranges but I thought an explicit `range` option actually offers more specific use cases for ranges that are not bound to the column type